### PR TITLE
CI: clean up Chromedriver and references from repo.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR removes dependencies and references to Chromedriver from the repository.

Key changes:
- removal of references to `CHROMEDRIVER_SKIP_DOWNLOAD` and `CHROMEDRIVER_SKIP_DOWNLOAD` env vars;
- removal of dependency on `chromedriver@88.x.x` from yarn;
- discontinue installation of Chrome in the docker image.

We no longer use Selenium in this repository as evidenced by https://github.com/Automattic/wp-calypso/pull/63108. Since Playwright manages its own browser installations, it is safe to remove Chrome and Chromedriver from the generated docker image.

#### Testing instructions

Ensure the following:
  - [ ] Gutenberg E2E (desktop)
  - [ ] Gutenberg E2E (mobile)
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)
  - [ ] Pre-Release E2E
  - [x] P2 E2E

Related to https://github.com/Automattic/wp-calypso/pull/63108

### CODE REVIEW

**Comment by noahtallen** [comment]
Why did we need it before? Any risks?

Edit: I also don't know if the mentioned TeamCity builds will utilize the base image changes here. IIRC, the branch-level docker image build will utilize the daily base image build, which doesn't include the changes to the base image here

**Comment by worldomonation** [comment]
> Why did we need it before? Any risks?

Ah, I should have clarified, sorry - Chrome was installed solely for Selenium tests. 
Now that Selenium has been entirely removed from the repository and no remaining jobs exist, I feel it is safe to remove it.

> the branch-level docker image build will utilize the daily base image build

My understanding of our pipelines is that:

- Docker base image is generated once a day using the Dockerfile.base that is in `trunk` at that time.
- once the build is complete, the images are uploaded and tags updated.
- subsequent PRs cut from `trunk` will use the new image, while older PRs will continue to use the older images.

Essentially, my understanding is that changes to `Dockerfile.base` will eventually propagate out to all branches. Is this not the case?

**Comment by noahtallen** [comment]
Yes, that's the case! I just meant that we would have to take extra manual steps to test the best build in this branch before merging

**Review by noahtallen — APPROVED** [review]
The change makes sense to me, given it's not needed anywhere!

**Comment by worldomonation** [comment]
@noahtallen Just a heads up that I expanded the scope of this PR quite a bit.

Instead of only discontinuing the installation of Chrome browser at the Dockerfile level, the revised PR:
- removes dependency on `chromedriver` package from the repository;
- removes chromedriver-related environment variables such as `CHROMEDRIVER_SKIP_DOWNLOAD` and `DETECT_CHROMEDRIVER_VERSION`;
- updates documentation to remove references to chromedriver;
- removes the chromedriver part of calypso-doctor.

I'm putting the merge of this PR on hold since the change is rather extensive and see if you'd like to have a look again.

File: test/e2e/docs/troubleshooting.md
Comment (noahtallen) [inline test/e2e/docs/troubleshooting.md:68]: Are we able to remove this too yet?

**Review by noahtallen — APPROVED** [review]
Nice! These changes also look good to me, trusting that the e2e tests don't need it

**Comment by noahtallen** [comment]
Is the failing mobile e2e related? it looked like it might possibly be, but I'm not sure

**Comment by worldomonation** [comment]
> Is the failing mobile e2e related? it looked like it might possibly be, but I'm not sure

It shouldn't be - the `page.screenshot: Target crashed` error is observed intermittently and I am not sure why.

I do plan to upgrade the Playwright version soon, hopefully it will go away with that.

File: test/e2e/docs/troubleshooting.md
Comment (worldomonation) [reply test/e2e/docs/troubleshooting.md:68]: I don't actually know where Puppeteer is used in `wp-calypso` and in my local environment I've always sidestepped the issue by setting that env var to `true`.

I'll do a quick check and if it's not too big of a change, will include it in this PR.

Comment (noahtallen) [reply test/e2e/docs/troubleshooting.md:68]: I think I might remember. It could be pulled in via `@wordpress/scripts` or some of the wordpress monorepo helpers

Comment (worldomonation) [reply test/e2e/docs/troubleshooting.md:68]: Searched for `puppeteer` and the only references I can find are:
- docs
- in yarn.lock file

Since we no longer run the Visual Regression suite (which used puppeteer) I think it might be safe to remove it?

Comment (worldomonation) [reply test/e2e/docs/troubleshooting.md:68]: (additionally, the version of Puppeteer that's in `yarn.lock` is too old to support arm64 macOS)

Comment (worldomonation) [reply test/e2e/docs/troubleshooting.md:68]: In any case, let's leave it for now and we can revisit Puppeteer in another patch.

**Comment by worldomonation** [comment]
Looks like the Code Style task is failing but this is a failure introduced in trunk.

**Pull request merged**
